src/core_plugins/timelion/public/directives/partials/suggestion.html

```diff
@@ -5,7 +5,7 @@
   <div
     class="suggestion"
     data-suggestion-list-item
-    ng-click="completeExpression($index)"
+    ng-click="onClickSuggestion({ suggestionIndex: $index })"
```

Comment:
This is a separate directive, so it accepts a handler function via an attribute which gets called when it's clicked.  It's analogous to how React components accept callbacks as props.